controllers/users/authenticate-web.js

```diff
@@ -21,26 +22,44 @@ const authenticateWeb = async (req, res) => {
   /**
    * @type {RegisterBody}
    */
-  const users = req.body;
+  const {exchangeToken} = req.body;
 
-  const checkUser = await User.findOne({
-    where: {
-      human_id: users.human_id
-    }
-  });
+  const verifyTokenResponse = await HumanIdService.verifyExchangeToken(exchangeToken);
+  if (!verifyTokenResponse.success) {
+    return res.status(500).json({
+      status: 'error on verify exchange token',
+      code: 500
+    });
+  }
+
+  const {data} = verifyTokenResponse || {};
+  const {appUserId} = data || {};
+
+  if (!appUserId) {
+    return res.status(500).json({
+      status: 'App user id is required',
+      code: 500
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The extraction of `exchangeToken` from the request body and its verification using `HumanIdService.verifyExchangeToken` is a good approach for token-based authentication. However, the error status code returned when the token verification fails or `appUserId` is not present should be 401 (Unauthorized) instead of 500 (Internal Server Error), as these cases represent an unauthorized access attempt.
```diff
-     return res.status(500).json({
+     return res.status(401).json({
        status: 'error on verify exchange token',
-       code: 500
+       code: 401
      });
...
-     return res.status(500).json({
+     return res.status(401).json({
        status: 'App user id is required',
-       code: 500
+       code: 401
      });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->